### PR TITLE
Person Details Improvements

### PR DESCRIPTION
**Pull request by BrandyBecker**
[Person Details Improvements Trello Card](https://trello.com/c/Id16IHps/55-fc-person-details-improvements)
------------------
### Description:
On the person details screen (case -> person)...

1) Give the add engagements icons at the top of the "Engagements" tab the same spacing as the same type of icons in the "Documents" tab. 
2) The words on the add engagements icons are not touchable. Make them touchable like the words under the icons in the "documents" tab. 
3) In the details tab, add some padding above the start of the details list. 
4) In the details tab, do not show the gender if it is "unspecified"
5) In the social media section, do not show the row if the value is empty (contains only white space. Use the trim() method to check)
6) In the details tab, show a check mark icon to the left of the emails, residences, and telephone numbers that are "verified". 
------------
### Committed Features
- Gives Engagement Icons same spacing as those in Documents tab
- Makes words on Engagement Icons touchable
- Adds padding above start of details list
- Won't show gender if it is "Unspecified"
- Won't show row if the value is empty / contains white-space
- Shows a check mark icon next to Verified emails, residences and telephone numbers
-------------
padding above details screen
unspecified gender
verify

### CODE REVIEW

File: src/screens/MoreScreen.tsx [hunk 1]
```diff
@@ -178,22 +213,90 @@ const MoreScreen = (props: Props) => {
...
+                                            }}
+                                        >
+                                            <View style={styles.modal}>
+                                                <Text>
+                                                    Error while logging out!
+                                                </Text>
+                                                <TouchableOpacity
+                                                    style={styles.saveButton}
+                                                    onPress={() => {
+                                                        dispatch({
```
Comment (ccannell) [inline src/screens/MoreScreen.tsx:249]: Use the enum for `LOG_OUT_MODAL_CLOSED` or make a function in `authActions.ts` that does it and call that.

File: src/screens/MoreScreen.tsx [hunk 2]
```diff
@@ -178,22 +213,90 @@ const MoreScreen = (props: Props) => {
...
+                                onRequestClose={() => {
+                                    console.log('Modal has been closed.');
+                                }}
+                            >
+                                <View style={styles.modal}>
+                                    <Text>Successfully logged out!</Text>
+                                    <TouchableOpacity
+                                        style={styles.saveButton}
+                                        onPress={() => {
+                                            dispatch({
```
Comment (ccannell) [inline src/screens/MoreScreen.tsx:287]: Use the enum for `LOG_OUT_MODAL_CLOSED` or make a function in `authActions.ts` that does it and call that.

Comment (BrandyBecker) [reply src/screens/MoreScreen.tsx:287]: I'm kind of confused as to what you're saying - I'm currently Dispatching `LOG_OUT_MODAL_CLOSED` which is also in tha `authActions.ts` as an `enum` - could you clarify a bit please?

Comment (BrandyBecker) [reply src/screens/MoreScreen.tsx:287]: image

Comment (BrandyBecker) [reply src/screens/MoreScreen.tsx:287]: `authActions`
image

`authReducer`
image

Comment (ccannell) [reply src/screens/MoreScreen.tsx:287]: You can add the following method to `authActions.tsx` and then call it from your code.

```
export const setLogoutModalClosed = (): ThunkResult<void> => (dispatch) => {
    dispatch({ type: AuthTypes.LOG_OUT_MODAL_CLOSED });
};
```

**Comment by ccannell** [comment]
@BrandyBecker FYI: You have merge conflicts that need to be resolved before this PR can merge.

**Comment by ccannell** [comment]
@BrandyBecker There are a couple small errors when building. Run `npm run build` and `npm run lint` locally to find and fix them.

**Comment by BrandyBecker** [comment]
> @BrandyBecker There are a couple small errors when building. Run `npm run build` and `npm run lint` locally to find and fix them.

I believe the build is passing now! It was a few empty lines throwing errors for prettier

**Comment by ccannell** [comment]
@BrandyBecker I merged master into your branch because I made the build checks stricter in master since this branch was created. I just want to ensure build doesn't fail when it merges into master.

**Pull request merged**